fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -1030,13 +1030,9 @@ ACTOR Future<bool> checkExclusion(Database db,
 					// Check if we are excluding a process that serves the storage role. If this check was true once, we don't have to check any further
 					// since we don't case in this variable about the count of excluded storage servers but only about if we exclude any storage server with the
 					// provided addresses.
-					if (!excludedAddressesContainsStorageRole) {
-						for (auto exclusion : addresses) {
-							if (exclusion.excludes(addr)) {
-								excludedAddressesContainsStorageRole = true;
-								break;
-							}
-						}
+					if (!excludedAddressesContainsStorageRole && excluded) {
+						excludedAddressesContainsStorageRole = true;
```

Comment:
It appears to me you would want to use the result of "addressExcluded(*exclusions, addr)" (by caching its result in a variable) instead of using "excluded" (as "excluded" would be true if this storage was already excluded).